src/core.js

```diff
@@ -332,19 +331,59 @@ const Editable = module.exports = class Editable {
     }
   }
 
-  // Highlight text within an editable.
-  //
-  // The first occurrence of the provided 'text' will be highlighted.
-  //
-  // The markup used for the highlighting will be removed from
-  // the final content.
-  //
-  // @param editableHost {DomNode}
-  // @param text {String}
-  // @param highlightId {String} Optional
-  //   Added to the highlight markups in the property `data-word-id`
-  highlight ({editableHost, text, highlightId}) {
-    return highlightSupport.highlightText(editableHost, text, highlightId)
+  /**
+   * Highlight text within an editable.
+   *
+   * By default highlights all occurences of `text`.
+   * Pass it a `textRange` object to highlight a
+   * specific text portion.
+   *
+   * The markup used for the highlighting will be removed
+   * from the final content.
+   *
+   *
+   * @param  {Object} options
+   * @param  {DOMNode} options.editableHost
+   * @param  {String} options.text
+   * @param  {String} options.highlightId Added to the highlight markups in the property `data-word-id`
+   * @param  {Object} [options.textRange] An optional range which gets used to set the markers.
+   * @param  {Number} [options.textRange.start]
+   * @param  {Number} [options.textRange.end]
+   * @return {Number} The text-based start offset of the newly applied highlight or `-1` if the range was considered invalid.
+   */
+  highlight ({editableHost, text, highlightId, textRange}) {
+    if (!textRange) {
+      return highlightSupport.highlightText(editableHost, text, highlightId)
+    }
+    if (typeof textRange.start !== 'number' || typeof textRange.end !== 'number') {
+      error(
+        'Error in Editable.highlight: You passed a textRange object with invalid keys. Expected shape: { start: Number, end: Number }'
+      )
+      return -1
+    }
+    if (textRange.start === textRange.end) {
+      error(
+        'Error in Editable.highlight: You passed a textRange object with equal start and end offsets, which is considered a cursor and therefore unfit to create a highlight.'
+      )
+      return -1
+    }
+    return highlightSupport.highlightRange(editableHost, text, highlightId, textRange.start, textRange.end)
+  }
+
+  /**
+   * Extracts positions of all DOMNodes that match `[data-word-id]`.
+   *
+   * Returns an object where the keys represent a highlight id and the value
+   * a text range object.
+   *
+   * @param  {Object} options
+   * @param  {DOMNode} options.editableHos
+   * @return {Object} ranges
```

Comment:
Why not include the text in the ranges? Otherwise how do you get the text of a range?

Reply:
> Why not include the text in the ranges? 
Done. Good point. Will have to adjust specs in the framework as well. JFYI